src/vcf/record.cpp

```diff
@@ -257,7 +259,7 @@ namespace ebi
                 if (key_values[ID] == field.first) {
                     found_in_meta = true;
                     try {
-                        check_field_cardinality(field.second, values, key_values[NUMBER]);
+                        check_info_field_cardinality(field.first, values, key_values[NUMBER]);
```

Comment:
Were we missing all these errors before? Are new tests added to check it?

Reply:
this `field.first` was used just for the human readable part in the report. instead of something like "Field GT was invalid as 0/0" something like "Field 0/0 was invalid as 0/0". I don't think it mattered a lot, but I will review this as I think that parameter is not needed inside `check_info_field_cardinality` anymore

Reply:
yep, it was not used anyway

Reply:
and the files that check this (wrong cardinality in INFO fields defined in the header) are test/input_files/v4.3/failed/failed_body_info_029.vcf, 30 and 31.